docs/examples/production-ready/rabbitmq.yaml

```diff
@@ -20,6 +20,18 @@ spec:
   persistence:
     storageClassName: ssd
     storage: "500Gi"
+  affinity:
+    podAntiAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+              - key: app.kubernetes.io/name
+                operator: In
+                values:
+                - production-ready
+          topologyKey: kubernetes.io/hostname
```

Comment:
The `topologyKey` is indented one level too far, causing the validation to fail if this example is deployed.

Reply:
Can you please confirm the issue? According to my Kubernetes cluster (v1.20.8) both `labelSelector` and `topologyKey` should be on the same level (i.e. under `requiredDuringSchedulingIgnoredDuringExecution`). In the file I commited both `labelSelector` and `topologyKey` are indented by 10 spaces.

If you want to confirm that on your cluster run for example `kubectl explain deployment.spec.template.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution`

Reply:
Something is definitely wrong:
```
$ kubectl apply -f https://raw.githubusercontent.com/rabbitmq/cluster-operator/15b3d20bf924b02a68d4a9bda7afa9d7f84b9271/docs/examples/production-ready/rabbitmq.yaml                    ⎈ kind-kind/examples
error: error validating "https://raw.githubusercontent.com/rabbitmq/cluster-operator/15b3d20bf924b02a68d4a9bda7afa9d7f84b9271/docs/examples/production-ready/rabbitmq.yaml": error validating data: [ValidationError(RabbitmqCluster.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution[0]): unknown field "podAffinityTerm" in com.rabbitmq.v1beta1.RabbitmqCluster.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution, ValidationError(RabbitmqCluster.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution[0]): unknown field "weight" in com.rabbitmq.v1beta1.RabbitmqCluster.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution, ValidationError(RabbitmqCluster.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution[0]): missing required field "topologyKey" in com.rabbitmq.v1beta1.RabbitmqCluster.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution]; if you choose to ignore these errors, turn validation off with --validate=false
```

Reply:
Turns out changing from `preferred` to `required` isn't just some cosmetics. Obviously weight does not play a role when we require something. I have commited a fix and I can apply it on v1.20.8.